system-addon/lib/ASRouter.jsm

```diff
@@ -139,10 +267,14 @@ class _ASRouter {
     switch (action.type) {
       case "CONNECT_UI_REQUEST":
       case "GET_NEXT_MESSAGE":
+        // Wait for our initial message loading to be done before responding to any UI requests
+        await this.messagesLoadedPromise;
+        // Check if any updates are needed first
+        await this.loadMessagesFromAllProviders();
         await this.sendNextMessage(target);
         break;
       case "BLOCK_MESSAGE_BY_ID":
-        this.setState(state => {
+        await this.setState(state => {
```

Comment:
how come these need to be awaited now? or were they always supposed to be awaited and you're fixing it?